src-tauri/src/tor_adapter.rs

```diff
@@ -50,22 +97,45 @@ impl ProcessAdapter for TorAdapter {
         let working_dir_string = convert_to_string(working_dir)?;
         let log_dir_string = convert_to_string(log_dir.join("tor.log"))?;
 
-        let args: Vec<String> = vec![
+        let mut args: Vec<String> = vec![
             "--allow-missing-torrc".to_string(),
+            "--ignore-missing-torrc".to_string(),
             "--clientonly".to_string(),
             "1".to_string(),
             "--socksport".to_string(),
             self.socks_port.to_string(),
             "--controlport".to_string(),
             format!("127.0.0.1:{}", self.control_port),
+            "--HashedControlPassword".to_string(),
+            EncryptedKey::hash_password(&self.password).to_string(),
             "--clientuseipv6".to_string(),
             "1".to_string(),
             "--DataDirectory".to_string(),
             working_dir_string,
             "--Log".to_string(),
             format!("notice file {}", log_dir_string),
         ];
+
+        if self.config.use_bridges {
+            let mut lyrebird_path = binary_version_path.clone();
+            lyrebird_path.pop();
+            lyrebird_path.push("pluggable_transports");
+            lyrebird_path.push("lyrebird");
+
+            args.push("--ClientTransportPlugin".to_string());
+            args.push(format!("obfs4 exec {} -enableLogging managed", convert_to_string(lyrebird_path)?));
+
+            args.push("--Bridge".to_string());
+            args.push("obfs4 177.235.180.145:8668 3AC17E97D18ADF92E9662D00EDE3B8FCCA6A6D80 cert=/WbHxF7lntn60mpieMRQ17YF88WfezgIzU0U+vQ3GMyCQUesVWtRkowZStfTO9cv6FutPw iat-mode=0".to_string());
```

Comment:
Let's do a call to the web to get the latest bridges

Reply:
Currently It's provided by users, from the form